### PR TITLE
feat: uniform scroll behavior and simplify layout layers

### PR DESCRIPTION
**Pull request by mcasimir**
Extracted from #3508: uniform scroll behavior

Also contains a lot of semi-related changes and drive-by fixes since this was a preparation to change backgrounds everywhere.

- clean up of backgrounds: 
  - base background colors are set on a container inside compass home, wired to the theme and any override previously present is removed or set to transparent
- Remove `Toolbar`, and give a function to `WorkspaceContainer`:  with the new lg style the `WorkspaceContainer` and `Toolbar` components lost most of their value (would both basically be just. a `div`). I've ditched  `Toolbar` entirely and changed `WorkspaceContainer` to take a `toolbar` property and take care of the scroll feature with shadow.
- Simplify top level layouts with css-grids (thanks @gribnoysup)
- Remove injectGlobalStyles, that wasn't actually working/doing anything
- Delete the loading placeholder from the dom after the app is loaded
- WorkspaceContainer is used everywhere to make the scrolling behavior consistent
- Improve zIndex stacking and isolate Editor in its own stack

https://user-images.githubusercontent.com/334881/191992153-cfb11414-1dd0-4356-aedf-8e52c22bb150.mp4

### CODE REVIEW

File: packages/compass-components/src/components/workspace-container.tsx [hunk 1]
```diff
@@ -1,15 +1,70 @@
...
+
+const shadowStyles = css({
+  position: 'absolute',
+  top: 0,
+  display: 'block',
+  width: '100%',
+  height: shadowHeight,
+  borderTop: `1px solid`,
+  flex: 'none',
+  zIndex: 1,
```
Comment (gribnoysup) [inline packages/compass-components/src/components/workspace-container.tsx:40]: Even if the overlap is small, I'd suggests disable pointer events on the shadow to make sure it doesn't mess with clicking

```suggestion
  zIndex: 1,
  pointerEvents: 'none',
```

**Comment by gribnoysup** [comment]
Test failures look legit: seems like some tests can't get reference of the editor anymore?

File: packages/compass-components/src/components/workspace-container.tsx (hunk 1 above)
Comment (mcasimir) [reply packages/compass-components/src/components/workspace-container.tsx:40]: Ah, nice idea! didn't know this let the clicks through

File: packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx [hunk 2]
```diff
@@ -82,7 +81,7 @@ function UnthemedSavedPipelines({
 }: SavedPipelinesProps) {
   return (
     <div className={savedPipelinesStyles}>
-      <Toolbar className={toolbarStyles}>
+      <div className={toolbarStyles}>
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx:84]: Should this also be a `toolbar` in a `WorkspaceContainer` then? Same with QueryHistory.

**Review by Anemy — APPROVED** [review]
Nice! I dig the consistent shadows on all of them. A light suggestion and question. lgtm

File: packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx (hunk 2 above)
Comment (mcasimir) [reply packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx:84]: oh .. that would probably make sense, is the same layout right (something fixed on top and content that scrolls below)?

I thought we could unify this better with the way we intend to do modals though, and use `WorkspaceContainer` internally:

```
<InteractivePopover header="{stringOrElement}"></content>
```

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-builder-workspace.module.less [hunk 3]
```diff
@@ -7,37 +7,6 @@
 .pipeline-workspace {
   display: flex;
   flex-direction: column;
-  overflow-y: scroll;
   width: 100%;
   flex-grow: 1;
 }
-
-.pipeline-workspace-container-container {
```
[the remark this answers is not in the document]
Comment (mcasimir) [reply packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-builder-workspace.module.less:15]: Makes sense, thanks!

File: packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx (hunk 2 above)
Comment (Anemy) [reply packages/compass-aggregations/src/components/saved-pipelines/saved-pipelines.tsx:84]: Sounds good, we can do it later in the project, definitely not a blocker here.

**Pull request merged**